sdk/identity/azure-identity/azure/identity/aio/_credentials/shared_cache.py

```diff
@@ -9,21 +9,31 @@
 from ..._internal.shared_token_cache import NO_TOKEN, SharedTokenCacheBase
 from .._internal.aad_client import AadClient
 from .._internal.exception_wrapper import wrap_exceptions
+from .base import AsyncCredentialBase
 
 if TYPE_CHECKING:
     from typing import Any
     from azure.core.credentials import AccessToken
     from ..._internal.aad_client import AadClientBase
 
 
-class SharedTokenCacheCredential(SharedTokenCacheBase):
+class SharedTokenCacheCredential(SharedTokenCacheBase, AsyncCredentialBase):
     """Authenticates using tokens in the local cache shared between Microsoft applications.
 
     :param str username:
         Username (typically an email address) of the user to authenticate as. This is required because the local cache
         may contain tokens for multiple identities.
     """
 
+    async def __aenter__(self):
+        await self._client.__aenter__()
+        return self
+
```

Comment:
No __aexit__?

Reply:
It's [on the base class](https://github.com/Azure/azure-sdk-for-python/pull/9090/files#diff-cea05bc674ee44f34b7048c0978d46ffR16-R17).